src/pages/SignUp/SignUp.jsx

```diff
@@ -58,7 +60,7 @@ function SignUp() {
       await signUp(name, email, password);
       navigate('/');
     } catch (error) {
-      // TODO: show error to the user
+      setEmailExist(true);
```

Comment:
any exception will occur will show this message, which isn't the best since the user will that his email is registered even if its not so we need to find a better way to check email already in use

Reply:
if the signup API is not ready yet, we can skip this error validation or use a mock error response.
example 
```
const myCustomError = { 
  Status: 409
  Message: "Email is already exists"
};
then check: 
```suggestion
      catch (error) {
      if (error.status === 409){
               setShowEmailAlreadyExistsError(true);
      } 
```